### PR TITLE
Feature/1099/questions list

### PR DESCRIPTION
**Pull request by mxrcury**
Create questions list and updated question component

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/90138904/1bb3249f-5d75-41dc-bff0-b2a00efcfe4b

### CODE REVIEW

File: src/containers/questions-list/QuestionsList.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+const QuestionsList: FC<QuestionsListProps> = ({ items, setItems }) => {
+  const reorder = (
+    list: QuestionWithCategory[],
+    startIndex: number,
+    endIndex: number
+  ): QuestionWithCategory[] => {
+    const result = Array.from(list)
+    const [removed] = result.splice(startIndex, 1)
+    result.splice(endIndex, 0, removed)
```
Comment (Tolik170) [inline src/containers/questions-list/QuestionsList.tsx:31]: can you return it immediately here?

File: src/containers/questions-list/QuestionsList.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    setItems(reorderedItems)
+  }
+
+  return (
+    <DragDropContext onDragEnd={onDragEnd}>
+      <Droppable droppableId='draggable'>
+        {(provided: DroppableProvided) => (
+          <Box {...provided.droppableProps} ref={provided.innerRef}>
+            {items.map((item, i) => (
```
Comment (Tolik170) [inline src/containers/questions-list/QuestionsList.tsx:55]: can you move this map above, if yes, better move it

File: src/containers/questions-list/QuestionsList.tsx (hunk 1 above)
Comment (mxrcury) [reply src/containers/questions-list/QuestionsList.tsx:31]: no, i can't cause `.splice` method modifies original array

File: src/containers/questions-list/QuestionsList.tsx (hunk 2 above)
Comment (mxrcury) [reply src/containers/questions-list/QuestionsList.tsx]: refactored

**Review by Mav-Ivan — APPROVED** [review]

**Review by TSlashDreamy — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**